### ISSUE TITLE
MIT License question

### ISSUE THREAD

**Issue by Nikit-Kalach** [S7NetPlus/s7netplus#425]
MIT License

Hello, my name is Nikita and I want to ask a question about license that is used here - MIT. I'm working in Siemens and as a company we want to provide an application, that uses the S7NetPlus, for Universities (for free, but project it self is commercial). Can we distribute it freely, and as I understood we should paste this MIT License text everywhere (even in the project) - am I right, could you tell me please? Also should we include this license text in the contract with University?

**Comment by mycroes** [comment]
Hi Nikita,

The MIT license affects source code and modifications to the source code. If you use the NuGet package of S7NetPlus you can freely distribute it with your software without any mention. If you distribute the source code along with the application you are bound to distributing the license as well. Even if you would use the source code to compile your own derivative work you're not required to distribute the license, however the internal copy of the S7NetPlus code stored at Siemens should technically still be accompanied of the license.

In short: offering the source code requires inclusion of the license; offering binaries has no further requirements.

Kind regards,
Michael
